File: typescript/ExampleUtil.ts
export class ExampleUtil {
    static async output(llmChunk: any): Promise<void> {
        if (!llmChunk.isFinal) {
            if ('content' in llmChunk) {
                process.stdout.write(llmChunk.content);
                await new Promise(f => setTimeout(f, 15));
            }
        } else {
            console.log(`\n\n${JSON.stringify(llmChunk)}\n\n`);
        }
    }
}

File: python/example_util.py
import time

class ExampleUtil():
    @staticmethod
    def output(llm_chunk):
        if (not llm_chunk.isFinal):
            if (hasattr(llm_chunk, 'content')):
                print(llm_chunk.content, end='', flush=True)
                time.sleep(0.015)
        else:
            print(f"\n\n{llm_chunk}\n\n")

